src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -870,7 +872,19 @@ private void ScanSyntaxToken(ref TokenInfo info)
                         TextWindow.AdvanceChar();
                     }
 
-                    info.Text = TextWindow.GetText(intern: true);
+                    if (_errorCount++ > 100)
```

Comment:
@sharwell - note that this is not a count of all errors. This is a count of invalid character errors which are very rare in actual source files. The typical reason for getting these errors is accidental parsing of nontext files like binaries. You do not need to see to 50K invalid characters to realize something wet badly wrong.

If native used 200, lets use that.